smsdk/client_v0.py

```diff
@@ -117,12 +119,26 @@ def __init__(self, tenant, site_domain="sightmachine.io"):
         :type site_domain: :class:`string`
         """
 
-        self.tenant = tenant
-
-        # Handle internal configuration
-        self.config = {}
-        self.config["protocol"] = "https"
-        self.config["site.domain"] = site_domain
+        if "://" in tenant:  # Full URL provided eg. "https://demo.sightmachine.io"
+            protocol, domain = tenant.split("://")[0], tenant.split("://")[1]
+            self.tenant = domain.split(".")[0]
```

Comment:
Always best to limit regex/splits to the number of elements, otherwise the function will enumerate the full string.
```suggestion
            self.tenant = domain.split(".", 1)[0]
```

Reply:
Done.